tikv/raftstore/coprocessor_test.go

```diff
@@ -0,0 +1,395 @@
+package raftstore
+
+import (
+	"fmt"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/kvproto/pkg/pdpb"
+	"github.com/pingcap/tidb/tablecodec"
+	"github.com/pingcap/tidb/util/codec"
+	"github.com/stretchr/testify/assert"
+	"math"
+	"testing"
+)
+
+type testSplitCheckFsm struct {
+	stopped bool
+	recv    <-chan Msg
+	mailbox *mailbox
+}
+
+func (s *testSplitCheckFsm) isStopped() bool {
+	return s.stopped
+}
+
+func (s *testSplitCheckFsm) setMailbox(mb *mailbox) {
+	s.mailbox = mb
+}
+
+func (s *testSplitCheckFsm) takeMailbox() *mailbox {
+	mb := s.mailbox
+	s.mailbox = nil
+	return mb
+}
+
+func newSplitCheckFsm(cap int) (chan Msg, fsm) {
+	ch := make(chan Msg, cap)
+	r := &testSplitCheckFsm{
+		recv: ch,
+	}
+	return ch, r
+}
+
+func newTestRouter() (*router, chan Msg) {
+
+	ctrlSender, ctrlFsm := newSplitCheckFsm(10)
+	normarlSender, normalFsm := newSplitCheckFsm(10)
+	ctrlBox := newMailbox(ctrlSender, ctrlFsm)
+
+	normalScheCh := make(chan Msg, msgDefaultChanSize)
+	normalSche := &normalScheduler{sender: normalScheCh}
+
+	router := newRouter(ctrlBox, normalSche, normalSche)
+	normalBox := newMailbox(normarlSender, normalFsm)
+	router.register(1, normalBox)
+	return router, normarlSender
+}
+
+func newTestSplitCheckRegion() *metapb.Region {
+	region := &metapb.Region{
+		Id: 1,
+		Peers: []*metapb.Peer{
+			new(metapb.Peer),
+		},
+		RegionEpoch: &metapb.RegionEpoch{
+			ConfVer: 5,
+			Version: 2,
+		},
+		StartKey: codec.EncodeInt(tablecodec.TablePrefix(), int64(0)),
+		EndKey:   codec.EncodeInt(tablecodec.TablePrefix(), int64(0)),
```

Comment:
The end key must be greater than start key unless both of them are empty.
Here the initial start key should be `[]byte{'t'}`, the initial end key should be `[]byte{'u'}`.